### PR TITLE
Register functionality refactored

### PR DESCRIPTION
**Pull request by vlionescu**

### CODE REVIEW

File: magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import { validteDriveData } from '../../utils/registrationValidationService';
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js]: typo

File: magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+		this.setState({
+			[name]: value
+		})
+	};
+
+	handleProjects = evt => {
+		const { target } = evt;
+		const value = target.type === 'checkbox' ? target.checked : target.value;
+		const { name } = target;
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js]: const { target: {type, checked, value, name}} = evt;

File: magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+		const newProjectsState = { [ name ]: value };
+		this.setState( oldstate => ( {
+			projects: Object.assign( {}, oldstate.projects, newProjectsState ),
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js]: projects: {...oldstate.projects, [name]: value}

(not sure if this works, but it should)

File: magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		} ) );
+	};
+
+	validateAndSendData = (e) => {
+
+		e.preventDefault();
+
+		const { personalDrive, projects } = this.state;
+
+		const errorData = validteDriveData( { personalDrive, projects} );
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js]: typo (derived from the import)

File: magic-volunteers-front-end/src/components/register-form-components/personal-info.react.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+		const { errors, errorMessage } = errorData;
+		this.setState({
+			errors,
+			errorMessage
+		});
+
+		const { nameError, dobError, phoneError, countyError, cityError, addressError } = errors;
+
+		if ( !nameError && !dobError && !phoneError && !countyError && !cityError && !addressError ) {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/personal-info.react.js]: if (Object.keys(errors).every(err => !err)) {...}

File: magic-volunteers-front-end/src/components/register-form-components/personal-info.react.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+			this.props.handleNext( { name, dob, phone, county, city, address } );
+		}
+	};
+
+	constructor(props) {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/personal-info.react.js]: This could be deleted.

File: magic-volunteers-front-end/src/components/register-form-components/personal-info.react.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
+import React, { Component } from 'react';
+
+import { validatePersonalData } from '../../utils/registrationValidationService';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+
+const states = [
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/personal-info.react.js]: I think these are named "counties" in the back-end.

File: magic-volunteers-front-end/src/components/register-form-components/registration-result.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+				props.history.replace( '/dashboard' );
+			}, 2000);
+		}
+	}
+
+
+	renderSuccess() {
+		return (
+			<div>
+				<h3>Felicitari pentru inregistrare</h3>
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/register-form-components/registration-result.js]: Îți mulțumim pentru înregistrare!

File: magic-volunteers-front-end/src/components/registerForm.css [hunk 9]
```diff
@@ -41,3 +46,106 @@
...
+.progress .circle.active .label {
+    color: #FFF;
+    background: #0c95be;
+    box-shadow: inset 0 0 2px rgba(0,0,0,.2);
+}
+.progress .circle.active .title {
+    color: #0c95be;
+}
+
+*, *:after, *:before {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/registerForm.css]: These styles are applied throughout the app, they should be moved from this register-specific file.

File: magic-volunteers-front-end/src/components/registerForm.react.js [hunk 10]
```diff
@@ -1,101 +1,110 @@
...
+	};
+
+	setStep = ( step ) => {
+		this.setState( {
+			step
+		})
+	};
+
+	renderStep( step ) {
+		switch( step ) {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/registerForm.react.js]: I'll do something to fix indentation after the merge.

File: magic-volunteers-front-end/src/components/registerForm.react.js [hunk 11]
```diff
@@ -104,278 +113,47 @@ class RegisterForm extends Component {
...
+					}
 
-                <button className="mv-btn mv-btn-primary">Inregistrare</button>
-                <div className="mv-info-box">
-                    <p>
-                        Ai deja un cont? <Link to="/login">Click aici</Link> pentru logare.
-                    </p>
-                </div>
-            </form>
+                    {/*<button className="mv-btn mv-btn-primary">Inregistrare</button>*/}
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/registerForm.react.js]: Please delete commented out code.

File: magic-volunteers-front-end/src/components/registerForm.react.js [hunk 12]
```diff
@@ -420,11 +199,11 @@ function buildData( data ) {
         address: {
             city: data.city,
             county: data.county,
-            details: data.details,
+            details: data.address,
```
Comment (melaniam) [inline magic-volunteers-front-end/src/components/registerForm.react.js]: Can you please double check that this is correct?

File: magic-volunteers-front-end/src/utils/CredentialsUtils.js [hunk 13]
```diff
@@ -58,6 +58,17 @@ class CredentialsUtils {
             }
         } );
     }
+
+    static validateEmail( email, onSuccess, onError) {
+        validateEmail( email ).then( response => {
+            console.log(response);
```
Comment (melaniam) [inline magic-volunteers-front-end/src/utils/CredentialsUtils.js]: Please remove the console.log.

File: magic-volunteers-front-end/src/utils/apiServices.js [hunk 14]
```diff
@@ -1,7 +1,7 @@
 import CredentialsUtils from "./CredentialsUtils";
 
 // TODO: move to a config file
-const baseUrl = "https://magicvolunteers.tech/api/";
+const baseUrl = "http://localhost:3030/api/";
```
Comment (melaniam) [inline magic-volunteers-front-end/src/utils/apiServices.js]: Please revert this change. We'll really have to move this into its own config file.

File: magic-volunteers-front-end/src/utils/registrationValidationService.js [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+
+	} else if ( !emailValidator.test(email) ) {
+		errors.emailError = true;
+
+		errorMessage = 'Formatul emailului este invalid';
+
+	} else if ( !passwordValidator.test(password) ) {
+		errors.passwordError = true;
+
+		errorMessage = 'Parola nu e sucifent de puternica';
```
Comment (melaniam) [inline magic-volunteers-front-end/src/utils/registrationValidationService.js]: 'Parola nu este suficient de sigură'

File: magic-volunteers-front-end/src/utils/registrationValidationService.js [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+
+	let errorMessage = '';
+
+	const phoneValidator = /^(\+4|)?(07[0-8]{1}[0-9]{1}|02[0-9]{2}|03[0-9]{2}){1}?(\s|\.|\-)?([0-9]{3}(\s|\.|\-|)){2}$/;
+
+	const nameValidator = /^[A-Za-z]+$/;
+
+	const { name, dob, phone, county, city, address } = data;
+
+	if ( !name || !dob || !phone || !county || !city || !address ) {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/utils/registrationValidationService.js]: if (Object.keys(data).some(key => !key)) {...}

File: magic-volunteers-front-end/src/utils/registrationValidationService.js [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+			errors.experienceDetailsError = true;
+		}
+
+		errorMessage = 'Te rog completeaza toate campurile';
+	}
+
+	return { errors, errorMessage }
+};
+
+const validteReferenceData = ( data ) => {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/utils/registrationValidationService.js]: typo

File: magic-volunteers-front-end/src/utils/registrationValidationService.js [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+			errors.relationshipError = true;
+		}
+
+		errorMessage = 'Te rog completeaza toate campurile';
+	}
+
+	return { errors, errorMessage }
+};
+
+const validteDriveData = ( data ) => {
```
Comment (melaniam) [inline magic-volunteers-front-end/src/utils/registrationValidationService.js]: typo

File: magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js (hunk 3 above)
Comment (vlionescu) [reply magic-volunteers-front-end/src/components/register-form-components/drive-info.react.js]: this was not done by me it was already implemented, but i'll change it
